nlp_primitives/tests/test_count_string.py

```diff
@@ -0,0 +1,170 @@
+                                   match_whole_words_only=False)
+
+        answers = [2, 2, 7, 5]
+        self.compare(primitive, self.test_cases, answers)
+
+    def test_regex_with_ignore_case(self):
+        primitive = self.primitive("h.l.o", ignore_case=True,
+                                   ignore_non_alphanumeric=False,
+                                   is_regex=True,
+                                   match_whole_words_only=False)
+
+        answers = [4, 2, 7, 7]
+        self.compare(primitive, self.test_cases, answers)
+
+    def test_regex_with_ignore_non_alphanumeric(self):
+        primitive = self.primitive("h.l.o", ignore_case=False,
+                                   ignore_non_alphanumeric=True,
+                                   is_regex=True,
+                                   match_whole_words_only=False)
+
+        answers = [2, 4, 7, 6]
+        self.compare(primitive, self.test_cases, answers)
+
+    def test_regex_with_match_whole_words_only(self):
+        primitive = self.primitive("h.l.o", ignore_case=False,
+                                   ignore_non_alphanumeric=False,
+                                   is_regex=True,
+                                   match_whole_words_only=True)
+
+        answers = [2, 2, 2, 2]
+        self.compare(primitive, self.test_cases, answers)
+
+    def test_regex_with_all_other_parameters(self):
+        primitive = self.primitive("h.l.o", ignore_case=True,
+                                   ignore_non_alphanumeric=True,
+                                   is_regex=True,
+                                   match_whole_words_only=True)
+
+        answers = [4, 4, 2, 3]
+        self.compare(primitive, self.test_cases, answers)
+
+    def test_overlapping_regex(self):
+        primitive = self.primitive("(?=(a.*a))", ignore_case=True,
+                                   ignore_non_alphanumeric=True,
+                                   is_regex=True,
+                                   match_whole_words_only=False)
+        test_cases = pd.Series([
+            'aaaaaaaaaa',
+            'atesta aa aa a'
+        ])
+        answers = [9, 6]
+        self.compare(primitive, test_cases, answers)
+
+    def test_the(self):
+        primitive = self.primitive("the", ignore_case=True,
+                                   ignore_non_alphanumeric=False,
+                                   is_regex=False,
+                                   match_whole_words_only=False)
+        test_cases = pd.Series([
+            "The fox jumped over the cat",
+            "The there then"
+        ])
+
+        answers = [2, 3]
+        self.compare(primitive, test_cases, answers)
+
+    def test_nan(self):
+        primitive = self.primitive("the", ignore_case=True,
+                                   ignore_non_alphanumeric=False,
+                                   is_regex=False,
+                                   match_whole_words_only=False)
+        test_cases = pd.Series([np.nan,
```

Comment:
Might be good to sprinkle in a `None` and `pd.NA` just to confirm there are not problems with those values as well.